mmdet3d/visualization/local_visualizer.py

```diff
@@ -747,15 +748,15 @@ def add_datasample(self,
 
         if draw_pred and data_sample is not None:
             if 'pred_instances_3d' in data_sample:
-                pred_instances_3d = data_sample.pred_instances_3d
-                # .cpu can not be used for BaseInstancesBoxes3D
-                # so we need to use .to('cpu')
-                pred_instances_3d = pred_instances_3d[
-                    pred_instances_3d.scores_3d > pred_score_thr].to('cpu')
-                pred_data_3d = self._draw_instances_3d(data_input,
-                                                       pred_instances_3d,
-                                                       data_sample.metainfo,
-                                                       vis_task, palette)
+                if len(data_sample.pred_instances_3d) > 0:
+                    pred_instances_3d = data_sample.pred_instances_3d
+                    # .cpu can not be used for BaseInstancesBoxes3D
+                    # so we need to use .to('cpu')
+                    pred_instances_3d = pred_instances_3d[
+                        pred_instances_3d.scores_3d > pred_score_thr].to('cpu')
+                    pred_data_3d = self._draw_instances_3d(
```

Comment:
It is better to do the check and skip drawing in self._draw_instances_3d to reduce and avoid the modification in two places.

Reply:
The exception handling location has been changed to reflect your opinion.

Thank you for your kind comments.